freqtrade/freqtradebot.py

```diff
@@ -129,6 +131,34 @@ def _throttle(self, func: Callable[..., Any], min_secs: float, *args, **kwargs)
         time.sleep(duration)
         return result
 
+    def refresh_tickers(self, pair_list: List[str]) -> Dict:
+        """
+        Refresh tickers asyncronously and return the result.
+        """
+        # TODO: maybe add since_ms to use async in the download-script?
+        # TODO: only refresh once per interval ? *may require this to move to freqtradebot.py
+        # TODO: Add tests for this and the async stuff above
+         
+        ticker_interval = self.strategy.ticker_interval
+        interval_in_seconds = int(ticker_interval[:-1]) * 60
+
+        should_not_update = ((self._klines_last_fetched_time + interval_in_seconds +1) > round(time.time()))
```

Comment:
should we cache this per pair? 
assuming one pair's refresh fails - it would not be updated for the whole ticker interval ...

it may be that we need a "better" container than a simple dict for `_klines`, so we can store the last-updated there too, but i guess a 2nd dict with the refresh-times works just as well.

Reply:
yep, was thinking about enriching _klines but I think having another one (_klines_last_fetched_time) would be more readable. your comment about per pair is fair. gonna have _klines_last_fetched_time dict which keeps timestamp per pair